kong/cmd/utils/prefix_handler.lua

```diff
@@ -248,6 +248,21 @@ local function compile_conf(kong_config, conf_template, template_env_inject)
     }
   }
 
+  local kong_proxy_access_log = kong_config.proxy_access_log
+  if kong_proxy_access_log ~= "off" then
+    compile_env.proxy_access_log_enabled = true
+  end
+  if kong_proxy_access_log then
+    local _, custom_format = string.match(kong_proxy_access_log, "^(%S+)%s(%S+)")
+    if custom_format then
+      compile_env.proxy_access_log_custom = true
+
+      if not string.match(kong_config.nginx_http_log_format or "", "^" .. custom_format) then
+        log.debug(table.concat({"a custom access_log format is set: '", custom_format, "', but not defined"}))
+      end
```

Comment:
If the user sets a custom format but does not define it, kong cannot start, which is expected.

We print a debug log for this case.